src/Components/Web.JS/src/Platform/WebAssemblyResourceLoader.ts

```diff
@@ -12,14 +12,52 @@ export class WebAssemblyResourceLoader {
       credentials: 'include',
       cache: networkFetchCacheMode
     });
+
+    // While we can expect an ASP.NET Core hosted application to include the environment, other
+    // hosts may not. Assume 'Production' in that case.
+    const environment = bootConfigResponse.headers.get('Blazor_Environment') || 'Production';
     const bootConfig: BootJsonData = await bootConfigResponse.json();
     const cache = await getCacheToUseIfEnabled(bootConfig);
 
-    return new WebAssemblyResourceLoader(bootConfig, cache);
+    return new WebAssemblyResourceLoader(bootConfig, environment, cache);
   }
 
-  constructor (public readonly bootConfig: BootJsonData, private cacheIfUsed: Cache | null)
-  {
+  constructor (public readonly bootConfig: BootJsonData, public readonly applicationEnvironment: string, private cacheIfUsed: Cache | null) {
+  }
+
+  readConfigFilesAsync(): Array<ConfigResult>{
+    let results = new Array<ConfigResult>();
+
+    const configFiles =  this.bootConfig.config;
+    if (!configFiles) {
+      return results;
+    }
+
+    const defaultAppSettingsJson = 'appsettings.json';
+    if (configFiles && configFiles.indexOf(defaultAppSettingsJson) != -1) {
+      results.push({
+        name: defaultAppSettingsJson,
+        contentPromise: getConfig(defaultAppSettingsJson)});
+    }
+
+    const environmentAppSettingsJson = `appsettings.${this.applicationEnvironment}.json`;
+    if (configFiles && configFiles.indexOf(environmentAppSettingsJson) != -1) {
+      results.push({
+        name: environmentAppSettingsJson,
+        contentPromise: getConfig(environmentAppSettingsJson)});
+    }
+
+    return results;
```

Comment:
TBH, all the logic in this function feels overlong to me, and a bit repetitious. How about something like the following, which also adds case-insensitive comparisons (though I'll defer to you on whether that's desirable):

```ts
  readConfigFilesAsync(): Array<ConfigResult>{
    const envAppSettingsName = `appsettings.${this.applicationEnvironment}.json`;
    return Object.keys(this.bootConfig.config || {})
      .filter(name => eqInsensitive(name, 'appsettings.json') || eqInsensitive(name, envAppSettingsName))
      .map(name => ({ name, contentPromise: getConfig(name) }));

    function eqInsensitive(a: string, b: string) {
      return a.toLowerCase() === b.toLowerCase();
    }

    async function getConfig(file: string) {
      const response = await fetch(file, { method: 'GET', credentials: 'include', cache: networkFetchCacheMode });
      return response.arrayBuffer();
    }
  }
```

Reply:
I'm assuming here we don't care about the ordering of the results in the returned array. Or that the calling code can take care of consuming the array entries in the order it wants.